src/App.tsx

```diff
@@ -1,24 +1,248 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import './App.scss';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoNotification }
+  from './components/TodoNotification/TodoNotification';
+import {
+  createTodo,
+  getTodos, removeTodo, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
+import { filterTodos } from './Helpers';
+import { ErrorMessages } from './types/ErrorMessages';
+import { UpdateTodoArgs } from './types/UpdateTodoArgs';
 
-const USER_ID = 0;
+const USER_ID = 6795;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(FilterOptions.ALL);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [updatingTodosId, setUpdatingTodosId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState<boolean>(false);
```

Comment:
This flag can be used in the header component only.
If you move a submit handler to the header, you can move this flag too.

Reply:
I can't move this ones because I use them in other components too.

`  const [errorMessage, setErrorMessage] = useState<ErrorMessages | null>(null);
  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
  const [updatingTodosId, setUpdatingTodosId] = useState<number[]>([]);`